pkg/handler/collector/collector.go

```diff
@@ -42,9 +42,10 @@ var (
 	documentCollectors = map[string]Collector{}
 )
 
-func RegisterDocumentCollector(c Collector, collectorType string) {
+func RegisterDocumentCollector(ctx context.Context, c Collector, collectorType string) {
```

Comment:
Just one update. All the other `Register` functions are returning an error. So lets make this one also return an error to be consistent.